aten/src/ATen/native/RNN.h

```diff
@@ -19,5 +19,36 @@ DECLARE_DISPATCH(rnn_packed_fn, gru_packed_cudnn_stub);
 DECLARE_DISPATCH(rnn_packed_fn, rnn_tanh_packed_cudnn_stub);
 DECLARE_DISPATCH(rnn_packed_fn, rnn_relu_packed_cudnn_stub);
 
-}} // namespace at::native
 
+void check_device(const Tensor& input, const TensorList& params, const TensorList& hiddens) {
+  auto input_device = input.device();
+  bool input_device_is_cuda = input_device.is_cuda();
+
+  auto check_tensors = [&](const std::string& name, const Tensor& t) {
+    if (!t.defined()) return;
+    auto t_device = t.device();
+    bool t_device_is_cuda = t_device.is_cuda();
+    AT_CHECK((input_device_is_cuda && t_device_is_cuda) ||
+             (!input_device_is_cuda && !t_device_is_cuda),
```

Comment:
Can't you just `input_device == t.device()`?

Reply:
Also, this check is simply `input_device_is_cuda == t_device_is_cuda`

Reply:
If you're worried about the error messages, then pushing this checking code only to the cuDNN path will let you assume that the device should be a GPU.

Reply:
Thanks for the comments! I will use `checkAllSameGPU` instead of introducing a new utils function here.